kv/mock.go

```diff
@@ -37,3 +38,63 @@ func (m mockClient) WatchKey(_ context.Context, _ string, _ func(interface{}) bo
 
 func (m mockClient) WatchPrefix(_ context.Context, _ string, _ func(string, interface{}) bool) {
 }
+
+// MockCountingClient is a wrapper around the Client interface that counts the number of times its functions are called.
+// This is used for testing only.
+type MockCountingClient struct {
+	client Client
+
+	ListCalls        *atomic.Int32
+	GetCalls         *atomic.Int32
+	DeleteCalls      *atomic.Int32
+	CASCalls         *atomic.Int32
```

Comment:
These should probably be Uint32 since they're counters.

Reply:
[9e8cce4](https://github.com/grafana/dskit/pull/618/commits/9e8cce4a8d07084a6407ba100f82f178306b6e08)